src/Bicep.LangServer/Server.cs

```diff
@@ -62,12 +71,13 @@ public async Task RunAsync(CancellationToken cancellationToken)
             await server.WaitForExit;
         }
 
-        private static void RegisterServices(Func<IResourceTypeProvider> resourceTypeProviderBuilder, IServiceCollection services)
+        private static void RegisterServices(CreationOptions creationOptions, IServiceCollection services)
         {
             // using type based registration so dependencies can be injected automatically
             // without manually constructing up the graph
-            services.AddSingleton<IResourceTypeProvider>(services => resourceTypeProviderBuilder());
-            services.AddSingleton<IFileResolver, FileResolver>();
+            services.AddSingleton<IResourceTypeProvider>(services => creationOptions.ResourceTypeProvider ?? new AzResourceTypeProvider());
+            services.AddSingleton<IFileResolver>(services => creationOptions.FileResolver ?? new FileResolver());
+            services.AddSingleton<IWorkspace, Workspace>();
```

Comment:
This reminds me... Is everything working ok with multi-folder workspaces in VS code? (Just wondering - I don't see anything that would prevent it.)

Reply:
Good point, will test this.

Reply:
This just worked without having to do anything special.